components/AutoComplete.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useEffect, useState } from "react";
+import "tailwindcss/tailwind.css";
+
+interface AutoCompleteProps {
+  value?: string;
+  suggestions?: string[];
+  onChange?: (value: string) => void | undefined;
+  onSelect?: (value: string) => void | undefined;
+}
+
+const AutoComplete = ({
+  value = "",
+  suggestions = [],
+  onChange = () => {},
+  onSelect = () => {},
+}: AutoCompleteProps) => {
+  const [isFocused, setIsFocused] = useState(false);
+  const [selectedIndex, setSelectedIndex] = useState(-1);
+
+  const suggestionsSize = suggestions.length;
+  const currentValue = suggestions[selectedIndex] || value;
+
+  useEffect(() => {
+    setSelectedIndex(-1);
+  }, [value, suggestionsSize]);
+
+  const onKeyDown = (e: React.KeyboardEvent) => {
+    switch (e.key) {
+      case "ArrowUp": {
+        e.preventDefault();
+        setSelectedIndex(
+          selectedIndex - 1 < -1 ? suggestionsSize - 1 : selectedIndex - 1
+        );
+        break;
+      }
+      case "ArrowDown": {
+        e.preventDefault();
+        setSelectedIndex(
+          selectedIndex + 1 >= suggestionsSize ? -1 : selectedIndex + 1
+        );
+        break;
+      }
+      case "Enter": {
+        e.preventDefault();
+        (e.target as HTMLInputElement).blur();
+        onSelect(currentValue);
+        break;
+      }
+      default:
+    }
+  };
+
+  return (
+    <div className="relative">
+      <input
+        className={[
+          "border",
+          "border-gray-300",
+          "rounded-2xl",
+          "focus:border-blue-300",
+          "focus:shadow",
+          "px-3 py-1",
+          "w-full",
+        ].join(" ")}
+        value={currentValue}
+        onChange={(e) => onChange(e.target.value)}
+        onFocus={() => setIsFocused(true)}
+        onBlur={() => setIsFocused(false)}
+        onKeyDown={onKeyDown}
+      />
+      {isFocused && !!suggestions.length && (
+        <div
+          className={[
+            "absolute",
+            "z-50",
+            "top-full",
+            "border-t",
+            "shadow",
+            "mt-1",
+            "w-full",
+          ].join(" ")}
+        >
+          {suggestions.map((s, i) => (
+            <div
+              key={s}
+              className={[
+                "border-b",
+                "px-3 py-1",
+                "cursor-pointer",
+                "bg-white",
+                "hover:bg-blue-100",
+                i === selectedIndex && "bg-blue-100",
+              ].join(" ")}
+              onMouseDown={() => onSelect(suggestions[i])}
```

Comment:
Cannot we use `s` directly?
```suggestion
              onMouseDown={() => onSelect(s)}
```